### PR TITLE
Feat/sonar mine

### PR DESCRIPTION
**Pull request by shirshanag**
<html>
<body>
<html><head></head><body>
<hr>
<h3>🔥 <strong>Pull Request: Sonar Rock vs Mine Classification using Logistic Regression</strong></h3>
<h4>📄 Summary</h4>
<p>This PR adds a complete machine learning project that predicts whether an object detected by sonar is a <strong>Rock (R)</strong> or a <strong>Mine (M)</strong> using <strong>Logistic Regression</strong>.<br>
The project includes data preprocessing, model training, evaluation, and testing on a sample input.</p>
<hr>
<h4>🧠 Key Features</h4>
<ul>
<li>
<p>Data loading and preprocessing using <strong>pandas</strong> and <strong>numpy</strong></p>
</li>
<li>
<p>Train-test split with <strong>stratified sampling</strong> for balanced data</p>
</li>
<li>
<p>Model training using <strong>Logistic Regression</strong> from <strong>scikit-learn</strong></p>
</li>
<li>
<p>Model performance evaluation on training and testing sets</p>
</li>
<li>
<p>Sample prediction for real-world testing</p>
</li>
<li>
<p>Added detailed <strong>README.md</strong> for documentation and reproducibility</p>
</li>
</ul>
<hr>
<h4>📊 Results</h4>

Metric | Training Accuracy | Test Accuracy
-- | -- | --
Logistic Regression | ~85% | ~78%

<p><em>(Results may slightly vary depending on random initialization)</em></p>
<hr>
<h4>🧩 Files Added</h4>
<ul>
<li>
<p><code inline="">sonar_prediction.py</code> → main ML pipeline</p>
</li>
<li>
<p><code inline="">Copy of sonar data.csv</code> → dataset</p>
</li>
<li>
<p><code inline="">README.md</code> → project overview, setup steps, and usage instructions</p>
</li>
</ul>
<hr>
<h4>🚀 Next Steps</h4>
<ul>
<li>
<p>Experiment with <strong>SVM</strong>, <strong>Random Forest</strong>, or <strong>Neural Networks</strong></p>
</li>
<li>
<p>Add <strong>data visualization</strong> for feature correlation analysis</p>
</li>
<li>
<p>Deploy using <strong>Flask</strong> or <strong>Streamlit</strong></p>
</li>
</ul>
<hr>
<h4>🙌 Acknowledgements</h4>
<p>Dataset Source: Kaggle – Sonar Dataset</a></p>
<hr>

### CODE REVIEW

**Comment by RK1905101** [comment]
hi @shirshanag, thankyou for contribution :)
